server/services/discovery/discovery_service.py

```
@@ -283,24 +331,48 @@ def run_discovery_for_user(user_id, connected_providers):
             enrichment_data["env_vars"] = serverless_result.get("env_vars", {})
             if serverless_result.get("errors"):
                 summary["errors"].extend(serverless_result["errors"])
+                # Serverless errors are provider-prefixed (e.g. "aws/fn-name"),
+                # attribute via substring match against the error string.
+                for err in serverless_result["errors"]:
+                    err_str = err if isinstance(err, str) else str(err)
+                    for pname in ("gcp", "aws", "azure"):
+                        if pname in connected_providers and pname in err_str.lower():
+                            summary["provider_errors"].setdefault(pname, []).append(err_str)
+                            break
             logger.info(f"[Discovery] Phase 2 Serverless enrichment complete")
```

Ruff (0.15.12)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)